src/pages/library/DataExplorer.js

```diff
@@ -1,30 +1,107 @@
-import { commonPaths } from 'src/components/breadcrumbs'
-import { Fragment } from 'react'
-import { div, h } from 'react-hyperscript-helpers'
+import _ from 'lodash/fp'
+import { Component, Fragment } from 'react'
+import { div, h, span } from 'react-hyperscript-helpers'
 import IframeResizer from 'react-iframe-resizer-super'
-import TopBar from 'src/components/TopBar'
+import { ajaxCaller } from 'src/libs/ajax'
+import { authStore, contactUsActive } from 'src/libs/state'
+import { centeredSpinner } from 'src/components/icons'
+import { commonPaths } from 'src/components/breadcrumbs'
+import { link } from 'src/components/common'
 import * as Nav from 'src/libs/nav'
 import * as Style from 'src/libs/style'
+import TopBar from 'src/components/TopBar'
+import * as Utils from 'src/libs/utils'
+
+// name must be name from Data Explorer dataset.json
+// authDomain must be authorization_domain from Data Explorer dataset.json
+const datasets = [
+  {
+    name: '1000 Genomes',
+    origin: 'https://test-data-explorer.appspot.com'
+  },
+  {
+    name: 'AMP PD - 2019_v1beta_0220',
+    origin: 'https://amp-pd-data-explorer.appspot.com',
+    authDomain: 'amp-pd-researchers'
+  },
+  {
+    name: 'Baseline Health Study',
+    origin: 'https://baseline-baseline-explorer.appspot.com',
+    authDomain: 'baseline-researchers'
+  },
+  {
+    name: 'Nurses\' Health Study',
+    origin: 'https://nhs-explorer.appspot.com',
+    authDomain: 'nhs_saturn_users'
+  },
+  {
+    name: 'UK Biobank',
+    origin: 'https://biobank-explorer.appspot.com',
+    authDomain: 'Kathiresan_UKBB'
+  }
+]
 
+const DataExplorer = _.flow(
+  ajaxCaller,
+  Utils.connectAtom(authStore, 'authState')
+)(class DataExplorer extends Component {
+  constructor(props) {
+    super(props)
+    const { dataset } = props
+    this.state = {
+      completedDeOauth: undefined,
+      groups: undefined
+    }
+    this.origin = datasets.filter(d => d.name === dataset)[0].origin
+    this.authDomain = datasets.filter(d => d.name === dataset)[0].authDomain
```

Comment:
Pre-computing instance variables like this creates 'pseudo-state' that can get out of sync and cause subtle bugs. Better to just compute on demand in `render`. Also, this could be shortened to e.g.
```
const { origin, authDomain } = _.find({ name: dataset }, datasets)
```